client/state/posts/selectors.js

```diff
@@ -90,11 +97,17 @@ export const getSitePost = createSelector(
  * @return {?Array}         Posts for the post query
  */
 export function getSitePostsForQuery( state, siteId, query ) {
-	if ( ! state.posts.queries[ siteId ] ) {
+	const manager = state.posts.queries[ siteId ];
+	if ( ! manager ) {
+		return null;
+	}
+
+	const posts = manager.getItems( query );
+	if ( ! posts ) {
 		return null;
 	}
 
-	return state.posts.queries[ siteId ].getItems( query );
+	return posts.map( ( post ) => getNormalizedPost( state, post.global_ID ) );
```

Comment:
curious to know why you call `getNormalizedPost` with the id instead of calling the normalize function since we have the post object already ?